src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs

```diff
@@ -341,7 +341,7 @@ private void AssertConnectionState(SubscriptionConnectionServerMessage connectio
                 if (connectionStatus.Type == SubscriptionConnectionServerMessage.MessageType.Error)
                     message += $". Exception: {connectionStatus.Exception}";
 
-                throw new SubscriptionMessageTypeException(message);
+                throw new Exception(message);
```

Comment:
Nope

Reply:
This will be handled after the release, reverts recent change

Reply:
why not add `SubscriptionMessageTypeException` to `ShouldTryToReconnect` swtich?

Reply:
Either that or let it inherit straight from Exception

Reply:
I think that for consistency it should inherit from `SubscriptionException` as other subscription specific exceptions. It needs to be correctly handled though.